include/swift/Basic/StringExtras.h

```diff
@@ -52,12 +52,21 @@ namespace swift {
   /// Determine the part of speech for the given word.
   PartOfSpeech getPartOfSpeech(StringRef word);
 
+  /// Copy \p string to \p Allocator and return it as a \c StringRef .
+  StringRef copyString(llvm::BumpPtrAllocator &Allocator, StringRef string);
+
+  /// Copy \p string to \p Allocator and return it as a null terminated C
+  /// string.
+  const char *copyCString(llvm::BumpPtrAllocator &Allocator, StringRef string);
+
```

Comment:
Makes sense. I should have done this when I used it in `SwiftSourceDocInfo`. What header in `SwiftSourceDocInfo.cpp` was actually including `CodeCompletion.h` 🤔